Psychtoolbox/PsychHardware/EyelinkToolbox/EyelinkDemos/SR-ResearchDemos/SimpleVideo/EyeLink_SimpleVideo.m

```diff
@@ -0,0 +1,396 @@
+        % Wait until user releases keys on keyboard:
+        KbReleaseWait;       
+        % Playback loop: Runs until end of movie or keypress:
+        while 1
+            % Check that eye tracker is  still recording. Otherwise close and transfer copy of EDF file to Display PC
+            error = Eyelink('CheckRecording');
+            if(error ~= 0)
+                fprintf('EyeLink Recording stopped!\n');
+                % Transfer a copy of the EDF file to Display PC
+                Eyelink('SetOfflineMode');% Put tracker in idle/offline mode
+                Eyelink('CloseFile'); % Close EDF file on Host PC
+                Eyelink('Command', 'clear_screen 0'); % Clear trial image on Host PC at the end of the experiment
+                WaitSecs(0.1); % Allow some time for screen drawing
+                % Transfer a copy of the EDF file to Display PC
+                transferFile; % See transferFile function below
+                cleanup; % Abort experiment (see cleanup function below)
+                return
+            end
+            % Wait for next movie frame, retrieve texture handle to it
+            tex = Screen('GetMovieImage', window, movie);
+            if tex<=0 % Valid texture returned? A negative value means end of movie reached
+                break;
+            end
+            % Draw the new texture immediately to screen:
+            Screen('DrawTexture', window, tex);            
+            % Update display:
+            Screen('Flip', window);
+            frameNum = frameNum + 1;
+            if frameNum == 1
+                % Write message to EDF file to mark the start time of stimulus presentation.
+                Eyelink('Message', 'STIM_ONSET');
+                % Write !V IAREA message to EDF file: creates interest areas in DataViewer
+                % See DataViewer manual section: Protocol for EyeLink Data to Viewer Integration > Interest Area Commands
+                Eyelink('Message', '!V IAREA RECTANGLE %d %d %d %d %d %s', 1, round(width/2-80), round(height/2-70), round(width/2+80), round(height/2+90), 'BOX_IA');
+                vidStart = GetSecs;  % Start a timer
+            end
+            % Write message to EDF file to mark the time of each video frame
+            Eyelink('Message', 'Frame to be displayed %d', frameNum);
+            % Write a !V VFRAME message to the data file specifying the frame number, location and file name so DataViewer can play back the video
+            Eyelink('Message', '%d !V VFRAME %d %d %d %s', 0, frameNum, round(width/2-Movx/2), round(height/2-Movy/2), movieName);
+            % End trial if space bar is pressed
+            [~, kbSecs, keyCode] = KbCheck;
+            if keyCode(spaceBar)
+                % Write message to EDF file to mark the space bar press time
+                Eyelink('Message', 'KEY_PRESSED');
+                timeOut = 'no';
+                % Release texture:
+                Screen('Close', tex);

```

Comment:
Might want to move this Close tex calls directly behind the DrawTexture call in line 269, this is less error prone if you don't need the tex anymore anyway after that draw.